ray-operator/controllers/ray/raycluster_controller_unit_test.go

```diff
@@ -2907,3 +2910,226 @@ func TestDeleteAllPods(t *testing.T) {
 	assert.Equal(t, 2, len(pods.Items))
 	assert.Subset(t, []string{"deleted", "other"}, []string{pods.Items[0].Name, pods.Items[1].Name})
 }
+
+// fakeEventRecorder mocks the record.EventRecorder interface. We can use the
+// record.FakeRecorder type as well, but that formats events into strings that
+// can then be read through a channel. That makes checking specific event properties
+// slightly tedious. fakeEventRecorder simply maintains a slice of events with some
+// filtering built on top.
+type fakeEventRecorder struct {
```

Comment:
This fake recorder is good, however, could we just use the `FakeRecorder` in the [kubernetes/client-go](https://github.com/kubernetes/client-go/blob/71959c526d543a5e4c3ca6fb808f535c2726483f/tools/record/fake.go#L29)?

Reply:
@rueian can you PTAL at the latest commit? If that approach looks good, I can squash the commits here. 🙏🏼